core/src/main/java/io/grpc/internal/AbstractManagedChannelImplBuilder.java

```diff
@@ -379,6 +385,75 @@ public T proxyDetector(@Nullable ProxyDetector proxyDetector) {
     return thisT();
   }
 
+  @Override
+  public T defaultServiceConfig(@Nullable Map<String, ?> serviceConfig) {
+    // TODO: use real parsing nameResolverHelper.parseServiceConfig() here or at build(),
+    defaultServiceConfig = checkMapEntryTypes(serviceConfig);
+    return thisT();
+  }
+
+  @Nullable
+  private static Map<String, ?> checkMapEntryTypes(@Nullable Map<?, ?> map) {
+    if (map == null) {
+      return null;
+    }
+    // Not using ImmutableMap.Builder because of extra guava dependency for Android.
+    Map<String, Object> parsedMap = new LinkedHashMap<>();
```

Comment:
`new LinkedHashMap<>(map.size());`

Reply:
The default load factor is 0.75, it will be rehashed.